app/jobs/clock/finalize_invoices_job.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module Clock
+  class FinalizeInvoicesJob < ApplicationJob
+    queue_as 'clock'
+
+    def perform
+      draft_invoices.each do |invoice|
+        Invoices::FinalizeService.call(invoice: invoice)
+      end
+    end
+
+    private
+
+    def draft_invoices
+      Invoice.draft.select { |i| Time.current > (i.created_at + grace_period(i).days) }
```

Comment:
Shouldn't we try to write this one in pure SQL. Here we are loading all draft invoices in memory, I feel it's not ideal performance-wise